packages/widgets/src/docklayout.ts

```diff
@@ -306,10 +306,10 @@ export class DockLayout extends Layout {
       mainConfig = null;
     }
 
-    // Create iterators over the old content.
-    let oldWidgets = this.widgets();
-    let oldTabBars = this.tabBars();
-    let oldHandles = this.handles();
+    // Create a snapshot of the old content.
```

Comment:
This used to work because the old `IIterator` instances were instantiated *before* `this._root` was set to `null` and afforded iteration even after line 215, but now that `this.tabBars()` is itself an iterator, it yields nothing after `this._root` has been set to `null`, so we must instead capture a snapshot of the old content before nullifying the root.